tests/MongoDB.Driver.Tests/OcspIntegrationTests.cs

```diff
@@ -0,0 +1,144 @@
+                        System.Threading.Thread.Sleep(2000);
+                    }
+                }
+            }
+        }
+
+        // private methods
+        /* We can't use DriverTestConfiguration.CreateDisposableClient because that results in a call to check
+         * the cluster type, which fails because a connection cannot be established to the mongod when testing
+         * the revoked certificate case */
+        private DisposableMongoClient CreateDisposableMongoClient(bool tlsInsecure)
+        {
+            var settings = DriverTestConfiguration.GetClientSettings().Clone();
+            settings.SslSettings.CheckCertificateRevocation = true;
+            // setting AllowInsecureTls= true will automatically set CheckCertificateRevocation to false
+            settings.AllowInsecureTls = tlsInsecure;
+            /* We want the heartbeat to fire so that we can get the HeartBeat exception in the cluster description
+             * in the exception that will be thrown when testing invalid certificates */
+            settings.HeartbeatInterval = TimeSpan.FromMilliseconds(500);
+            /* We lower the server selection timeout to speed up tests, but choosing a value <=5s may
+             * result in the driver being unable to perform OCSP endpoint checking in time, causing a
+             * ServerSelectionTimeout that does include a certificate revocation status error message. */
+            settings.ServerSelectionTimeout = TimeSpan.FromSeconds(5*2); // must be > 5s
+            // settings.SdamLogFilename = @"C:\temp\sdam" + $"{tlsInsecure}.log";
+
+            return new DisposableMongoClient(new MongoClient(settings));
+        }
+
+        private bool GetSecureClientExpectedOutcomeOrSkipTest()
+        {
+            var ocspOutcomeEnvironmentVariableValue
+                = Environment.GetEnvironmentVariable(_ocspOutcomeEnvironmentVariableName);
+            if (ocspOutcomeEnvironmentVariableValue == null)
+            {
+                throw new SkipException($"Skipping because {_ocspOutcomeEnvironmentVariableName} not set.");
+            }
+            if (!Boolean.TryParse(ocspOutcomeEnvironmentVariableValue, out var successExpected))
+            {
+               throw new Exception(
+                   $"Invalid value of {ocspOutcomeEnvironmentVariableValue} in {_ocspOutcomeEnvironmentVariableName}." +
+                   $" Expected true/false");
+            }
+            return successExpected;
+        }
+
+        /// <summary>
+        /// Sets up environment variables required for local testing.
+        /// </summary>
+        /// <param name="mockResponderValidatesAllCertificates">
+        /// Set to true if the mock OCSP responder will validate all certificates.
+        /// Set to false if the mock OCSP responder will revoke all certificates.
+        /// </param>
+        private void SetupLocalTestingEnvironmentVariables(bool mockResponderValidatesAllCertificates)
```

Comment:
not sure about this method, in case if we add sonar or styleCop later, this method will trigger warning

Reply:
Maybe it's ok just to mentioned expected connection string via comment?

Reply:
I'm fine with removing that when we do add sonar/StyleCop as I found commenting/uncommenting a single line to be far quicker than updating my IDE's environment variable settings, but admittedly this is personal preference. I can remove it if you feel strongly about it :)

Reply:
my point that we already had a similar situation in FLE (at least), and we didn't leave any local configuration in the final version.

Reply:
In the interest of time, I'll remove this :)